### PR TITLE
[GLJS-647] Fixed displaying of waypoints on the route

### PR DESCRIPTION
**Pull request by underoot**
- Added support of displaying route and instructions for all route lags, which means that missed waypoint functionality also was fixed.
- Added relevant integration test.
- Broken and noisy tests were updated to be clean and test more realistic scenario. For example check that instructions actually were rendered and [it's not just error message](https://github.com/mapbox/mapbox-gl-directions/blob/master/test/test.instructions.js#L24).
- Updated Mapbox GLJS to latest supported version
- Switched from deprecated "deep-assign" to "lodash.merge" in setOptions action
- Get rid off store and request singletons. Some assertions because of that had been launched couple of times

https://github.com/mapbox/mapbox-gl-directions/assets/2480721/92007312-52c8-4870-bad7-fbb8a737ffd8

### CODE REVIEW

File: src/actions/index.js [hunk 1]
```diff
@@ -68,7 +67,8 @@ function fetchDirections() {
     if (language) options.push('language='+language);
     if (exclude) options.push('exclude=' + exclude);
     if (accessToken) options.push('access_token=' + accessToken);
-    request.abort();
+
+    const request = new XMLHttpRequest();
```
Comment (underoot) [inline src/actions/index.js:71]: Probably after that will need to follow order of requests in callbacks to have ability anyway cancel requests and don't trigger long response updates

File: src/constants/action_types.js [hunk 2]
```diff
@@ -1,6 +1,7 @@
 export const DESTINATION = 'DESTINATION';
 export const DESTINATION_CLEAR = 'DESTINATION_CLEAR';
 export const DESTINATION_QUERY = 'DESTINATION_QUERY';
+export const DESTINATION_QUERY_START = 'DESTINATION_QUERY_START';
```
Comment (stepankuzmin) [inline src/constants/action_types.js:4]: Is it a leftover? I couldn't find any reference to it.

**Review by stepankuzmin — APPROVED** [review]
Good job! 👍

File: src/actions/index.js (hunk 1 above)
Comment (underoot) [reply src/actions/index.js:85]: [6b80678](https://github.com/mapbox/mapbox-gl-directions/pull/310/commits/6b80678fdf529ff43e7a686a3fcd1868224aaa1e)

**Pull request merged**